Source/Controllers/MainViewControllers/ConnectionView/SPConnectionController.m

```diff
@@ -2291,11 +2290,11 @@ - (void)initiateSSHTunnelConnection
 	
 	[sshTunnel setParentWindow:[dbDocument parentWindow]];
 
-	// Add keychain or plaintext password as appropriate - note the checks in initiateConnection.
-	if (connectionSSHKeychainItemName && !isTestingConnection) {
+    // Only set the password if there is no Keychain item set or the connection is being tested or the password is different than in Keychain.
+    if ((isTestingConnection || !connectionSSHKeychainItemName || (connectionSSHKeychainItemName && ![[self sshPassword] isEqualToString:@"SequelAceSecretPassword"])) && [self sshPassword]) {
+        [sshTunnel setPassword:[self sshPassword]];
```

Comment:
If [self sshPassword] is set to nil this if check will fail and fall through to the else which I think is unexpected behavior. I think we should get rid of the && [self sshPassword], or make a second check to see it's ![self sshPassword].
In other words, what happens if the password field is nulled out in the connection edit process (And there was previously  a password stored in the keychain).

Reply:
Sooo if you take a look at how it was, this didn't changed, except it allowed users to set password even for saved favorite (like for TCP).

Previously (on left side) if you were in favorite, it used keychain password, if you were not, but password did exist, we used that. If empty was nil, it was left nil.

Now it's the same - if you are testing connection, or you are not in saved favorite, or you are in saved favorite but password changed - use the password.

If there was previously a password, and you want to remove it, you need to hit save to remove it from keychain as well.